### PR TITLE
Fix Bug 7823 - Frame corner radius.

### LINKED ISSUE #7823: [Bug] Frame corner radius.
### Description
After updating from v3.6.0.344457 to v4.2.0.848062 frame cease to cut corners of child elements.
### Steps to Reproduce

1. Create frame with any corner radius and NOT ZERO padding.
2. Create one more frame as a child of the first with any corner radius and ZERO `padding.`
3. Into the second frame insert boxview(can be any view)

### Expected Behavior
Child elements has cut corners.
### Actual Behavior
Child elements don't has cut corners

### Code for screenshot

    <StackLayout>
        <Frame CornerRadius="5" BackgroundColor="Red" Padding="10">
            <Frame CornerRadius="10" BackgroundColor="Blue" Padding="0">
                <BoxView BackgroundColor="Green" HeightRequest="100" WidthRequest="100"/>
            </Frame>
        </Frame>
    </StackLayout>

### Screenshots
photo_2019-10-04_16-53-25

### PR DESCRIPTION
**Pull request by felipebaltazar**
### Description of Change ###
I created changed event, on "CornerRadiusProperty", to set "IsClippedToBounds" to true when radius is greater than 0.

### Issues Resolved ### 
- fixes #7823 

### API Changes ###
 None

### Platforms Affected ### 
- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

Frame will cut corners of child elements, when corner radius greater than 0 and padding equals 0. 

### Before/After Screenshots ### 

- Before

<p align="center">
	<kbd>
		before screenshot
	</kbd>
</p>

- After
<p align="center">
	<kbd>
		image
	</kbd>
</p>

### Testing Procedure ###

1. Create frame with any corner radius and NOT ZERO padding.
1. Create one more frame as a child of the first with any corner radius and ZERO padding.
1. Into the second frame insert boxview(can be any view)

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by felipebaltazar** [comment]
Hello!
I've created a android UI test, for this solution, any tips to create a test for IOS?
(important! i dont have a mac)

<p align="center">
	<kbd>
		Ui test passed
	</kbd>
</p>

The tag "retarget-branch-required" is to change to 4.2.0 and rebase my current branch?

**Comment by jfversluis** [comment]
> The tag "retarget-branch-required" is to change to 4.2.0 my pr base branch? because has so many commit's between theses branch's, i think is better create another PR and close this, in this case

Since it is a regression bug this should target 4.2 at this time, yes. You should not worry about the many differences in commits, you should be able to rebase on the 4.2 branch and just add your commits on top of that. Do you know how to do that?

As for the UITest, that is a bit of a problem if you don't have a Mac at this time. Make sure you target the right branch etc. we will figure something out for the iOS test.

Thanks for your first time contribution!

File: Xamarin.Forms.Core/Frame.cs [hunk 1]
```diff
@@ -87,5 +87,11 @@ void IBorderElement.OnBorderColorPropertyChanged(Color oldValue, Color newValue)
 		bool IBorderElement.IsBorderColorSet() => IsSet(BorderColorProperty);
 
 		bool IBorderElement.IsBorderWidthSet() => false;
+
+		static void OnCornerRadiusChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (bindable is Frame frame)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Frame.cs]: use direct cast

File: Xamarin.Forms.Core/Frame.cs [hunk 2]
```diff
@@ -87,5 +87,11 @@ void IBorderElement.OnBorderColorPropertyChanged(Color oldValue, Color newValue)
...
+				frame.IsClippedToBounds |= (float)newValue > 0f;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Frame.cs]: only set IsClippedToBounds to 'true' if it never was set by the user. do not reset it to false, as we do not know who set it to true

Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Frame.cs]: Or even better, avoid this side effect, and handle this in the android renderer

Comment (felipebaltazar) [reply Xamarin.Forms.Core/Frame.cs]: @StephaneDelcroix  Any tip to how detect user change?  I mean, i thinking to track `OnPropertyChanged` and keep this information in a variable (like `bool clippedToBoundsUserChanged`)

**Comment by felipebaltazar** [comment]
Thank you for your feedback! 🎉

> Do you know how to do that?
Yeah! Has so many conflicts on the way, but i really can do this 💪

> we will figure something out for the iOS test.
Thanks!🎉

File: Xamarin.Forms.Core/Frame.cs (hunk 2 above)
Comment (felipebaltazar) [reply Xamarin.Forms.Core/Frame.cs]: @StephaneDelcroix I found this solution: (applied at #8032 )
https://github.com/xamarin/Xamarin.Forms/pull/8032/files#diff-7e99600fa8a972c09f94b521218f1e0eR241

**Comment by felipebaltazar** [comment]
Hello!
I am closing this PR to deal with this #8032 
I`ve made the changes suggested in code review 🎉

**Pull request closed without merging**